manifests/tftp.pp

```diff
@@ -29,4 +29,75 @@
       root => $foreman_proxy::tftp_root,
     }
   }
+
+  case $::osfamily {
+    'RedHat': {
+      case $::operatingsystem {
+        'Fedora': {
+          if versioncmp($::operatingsystemrelease, '19') >= 0 or $::operatingsystemrelease == 'Rawhide' {
+            $grub_type = 'redhat'
+          } else {
+            $grub_type = 'redhat_old'
+          }
+        }
+        /^(RedHat|CentOS|Scientific|OracleLinux)$/: {
+          if versioncmp($::operatingsystemmajrelease, '7') >= 0 {
+            $grub_type = 'redhat'
+          } else {
+            $grub_type = 'redhat_old'
+          }
+        }
+      }
+    }
+    'Debian','Ubuntu': {
+      $grub_type = 'debian'
+    }
+  }
+
+  case $::operatingsystem {
+    'Fedora': {
+      $grub_efi_path = 'fedora'
+    }
+    'CentOS': {
+      $grub_efi_path = 'centos'
+    }
+    /^(RedHat|Scientific|OracleLinux)$/: {
+      $grub_efi_path = 'redhat'
+    }
+  }
+
+  $efi_dir = '/usr/lib/grub/x86_64-efi'
+  $grub_modules = 'all_video boot btrfs cat chain configfile echo efifwsetup efinet ext2 fat font gfxmenu gfxterm gzio halt hfsplus iso9660 jpeg loadenv loopback lvm mdraid09 mdraid1x minicmd normal part_apple part_msdos part_gpt password_pbkdf2 png reboot search search_fs_uuid search_fs_file search_label serial sleep syslinuxcfg test tftp video xfs linux backtrace usb usbserial_common usbserial_pl2303 usbserial_ftdi usbserial_usbdebug linuxefi'
+
+  case $grub_type {
+    'redhat': {
+      ensure_packages(['grub2-efi','grub2-efi-modules','grub2-tools','shim'], { ensure => "installed", })
+
+      exec {'build-grub2-efi-image':
+        command => "/usr/bin/grub2-mkimage -O x86_64-efi -d $efi_dir -o $foreman_proxy::tftp_root/grub2/grubx64.efi -p '' $grub_modules",
+        creates => "$foreman_proxy::tftp_root/grub2/grubx64.efi",
+        require => [File[$foreman_proxy::tftp_dirs], Package['grub2-tools']],
+      }
+
+      # copy and rename the file (can't use the helper function)
+      file {"$foreman_proxy::tftp_root/grub2/grubx64s.efi":
+        ensure  => file,
+        source  => "/boot/efi/EFI/$grub_efi_path/grubx64.efi",
+        require => File[$foreman_proxy::tftp_dirs],
+        #require     => $required_tftp_classes; -- is this required?
+      }
```

Comment:
The idea behind copying `grubx64s.efi` (stands for small) is to copy an alternative EFI loader for those that would prefer the distribution one. But maybe it's not necessary at all... I can remove that.